src/graph.rs

```diff
@@ -535,7 +536,9 @@ where
         resize_buffer_to(&mut self.dry_buffer, buffer_size);
 
         // Initialise all connection buffers.
-        for connection in self.dag.edge_weights_mut() {
+        let edge_ids = self.edge_references().map(|e| e.id()).collect::<Vec<_>>();
+        for edge_id in edge_ids {
+            let connection = self.connection_mut(edge_id).unwrap();
```

Comment:
Unfortunately, this also means that we cannot [mutably iterate the graph's nodes](https://github.com/petgraph/petgraph/issues/179) directly (yet). To address this, I have implemented a workaround which is slightly less efficient.